finality-aleph/src/sync/mock/backend.rs

```diff
@@ -254,7 +233,37 @@ impl Finalizer<MockJustification> for Backend {
 
 impl BlockImport<MockBlock> for Backend {
     fn import_block(&mut self, block: MockBlock) {
-        self.import(block.header);
+        if !block.verify() {
+            return;
+        }
+
+        let mut storage = self.inner.lock();
+
+        let parent_id = match block.parent_id() {
+            Some(id) => id,
+            None => return,
+        };
+
+        if storage.blockchain.contains_key(&block.id())
+            || !storage.blockchain.contains_key(&parent_id)
+            || block.id().number() != parent_id.number() + 1
+        {
+            return;
+        }
+
+        if block.id().number() > storage.best_block.number()
+            && is_predecessor(
```

Comment:
Does it mean that the sequence `best_block(t)` where `t` is time forms a chain? It's not true in the general case right? It's possible that the next best block will appear on a different branch. Is it just in these tests that  we assume such behavior?

Reply:
good catch, I guess before we really define what the best_block means to us this does not matter :P

Reply:
I've removed the whole best block logic, now the trait method just returns an error

we will reintroduce proper implementation when we know what exactly we want to mock...